Enable HPA by default (#27)
giantswarm/nginx-ingress-controller-app: CHANGELOG.md

and this project's packages adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [v1.6.0] 2020-02-26

### Changed

- Enabled HorizontalPodAutoscaler by default.
- Based on HPA trials done so far, following settings have been adjusted to better fit actual observed usage profiles:
  - CPU resource requests have been adjusted from 500m to 2 CPU
    - 0.5 CPU was not enough for all the processes NGINX Ingress Controller starts
  - Memory requests changed from 600Mi to 2.5GB
    - Scaling out does not shard Ingress definitions and other configurations stored in memory of every nginx-ingress-controller replica
    - Memory usage spikes during configuration reloads
    - It improves the HPA stability
  - Default number of nginx worker processes was changed from 4 to 1
    - This reduced memory usage of each replica
    - It didn't affect request handling capacity
    - Better defaults considering CPU requests and number of processes running on every nginx-ingress-controller replica.
- To avoid cluster-operator and HPA collision and nginx service disruption, this release also breaks with cluster-operator controllable nginx Deployment replicas count

>   - `ingressController.replicas` which was previously dynamically set buy cluster-operator is now removed

Reviewer: ```suggestion
  - `ingressController.replicas` which was previously dynamically set by cluster-operator is now removed
```

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project's packages adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [v1.6.0] 2020-02-21

### Changed

- Enabled HorizontalPodAutoscaler by default.
- Based on HPA trials done so far, following settings have been adjusted to better fit actual observed usage profiles:
  - CPU resource requests have been adjusted from 500m to 2 CPU
    - 0.5 CPU was not enough for all the processes NGINX Ingress Controller starts
  - Memory requests changed from 600Mi to 2.5GB

>     - Scaling out does not shard Ingress defitions and other configuration stored in memory of every NGINX Ingress Controller replica

Author: Thanks for spotting typos. For nginx IC references I think @cokiengchiara suggested us to use NGINX Ingress Controller consistently in release notes in future. I'm fine with either.

    - Memory usage spikes during configuration reloads;

>     - 2.5GB seems to be better default judging by HPA stability and number of Ingress resources defined on clusters used in HPA trials so far

Reviewer 2: ```suggestion
    - It improves the HPA stability 
```
WHat do you mean by `and number of Ingress resources defined on clusters `?

  - Default number of nginx worker processes was changed from 4 to 1
    - This reduced memory usage of each replica

>     - It didn't seem to affect request handling capacity in HPA trials

Reviewer 2: ```suggestion
    - It didn't affect request handling capacity 
```

>     - This seems like a better default considering CPU requests and number of processes running on every NGINX Ingress Controller replica.

Reviewer 2: ```suggestion
    - Better defaults considering CPU requests and number of processes running on every nginx-ingress-controller replica.
```
